src/lib/message/ExchangeContext.cpp

```diff
@@ -0,0 +1,1664 @@
+}
+
+void ExchangeContext::TimerT(System::Layer * aSystemLayer, void * aAppState, System::Error aError)
+{
+    ExchangeContext * client = reinterpret_cast<ExchangeContext *>(aAppState);
+
+    if ((aSystemLayer == NULL) || (aAppState == NULL) || (aError != CHIP_SYSTEM_NO_ERROR))
+    {
+        return;
+    }
+    if (client->StartTimerT() != CHIP_NO_ERROR)
+    {
+        nlLogError("EC: cant start T\n");
+    }
+}
+
+void ExchangeContext::TimerTau(System::Layer * aSystemLayer, void * aAppState, System::Error aError)
+{
+    ExchangeContext * ec = reinterpret_cast<ExchangeContext *>(aAppState);
+
+    if ((aSystemLayer == NULL) || (aAppState == NULL) || (aError != CHIP_SYSTEM_NO_ERROR))
+    {
+        return;
+    }
+    if (ec->msgsReceived < ec->rebroadcastThreshold)
+    {
+        ChipLogDetail(ExchangeManager, "Trickle re-send with duplicate message counter: %u", ec->msgsReceived);
+        ec->ResendMessage();
+    }
+    else
+    {
+        ChipLogDetail(ExchangeManager, "Trickle skipping this interval");
+    }
+    if ((ec->RetransInterval == 0) || (ec->RetransInterval <= ec->backoff))
+    {
+        return;
+    }
+    if (aSystemLayer->StartTimer(ec->RetransInterval - ec->backoff, TimerT, ec) != CHIP_NO_ERROR)
+    {
+        nlLogError("EC: cant start Tau\n");
+    }
+}
+
+bool ExchangeContext::MatchExchange(ChipConnection * msgCon, const ChipMessageInfo * msgInfo,
+                                    const ChipExchangeHeader * exchangeHeader)
+{
+    // A given message is part of a particular exchange if...
+    return
+
+        // The exchange identifier of the message matches the exchange identifier of the context.
+        (ExchangeId == exchangeHeader->ExchangeId)
+
+        // AND The message was received over the connection bound to the context, or the
+        // message was received over UDP (msgCon == NULL) and the context is not bound
+        // to a connection (Con == NULL).
+        && (Con == msgCon)
+
+        // AND The message was received from the peer node associated with the exchange, or the peer node identifier is 'any'.
+        && (((PeerNodeId == kAnyNodeId) && (msgInfo->DestNodeId == ExchangeMgr->FabricState->LocalNodeId)) ||
+            (PeerNodeId == msgInfo->SourceNodeId))
+
+        // AND The message was sent by an initiator and the exchange context is a responder (IsInitiator==false)
+        //    OR The message was sent by a responder and the exchange context is an initiator (IsInitiator==true) (for the broadcast
+        //    case, the initiator is ill defined)
+
+        && (((exchangeHeader->Flags & kChipExchangeFlag_Initiator) != 0) != IsInitiator());
+}
+
+/**
+ *  Handle trickle message within the exchange context.
+ *
+ *  @param[in]    pktInfo    A pointer to the IPPacketInfo object.
+ *
+ *  @param[in]    msgInfo    A pointer to the CHIP message info structure.
+ *
+ */
+void ExchangeContext::HandleTrickleMessage(const IPPacketInfo * pktInfo, const ChipMessageInfo * msgInfo)
```

Comment:
What's a trickle message?

Reply:
Per my other comment, we should remove the references to the this functionality